packages/noya-designsystem/src/components/ColorInputField.tsx

```diff
@@ -1,30 +1,39 @@
-import type Sketch from '@sketch-hq/sketch-file-format-ts';
+import Sketch from '@sketch-hq/sketch-file-format-ts';
 import { ForwardedRef, forwardRef, useMemo } from 'react';
 import styled from 'styled-components';
+import { getGradientBackground } from '../utils/getGradientBackground';
 import { sketchColorToRgbaString } from '../utils/sketchColor';
 
-const Container = styled.button(({ theme, color }) => ({
-  outline: 'none',
-  width: '40px',
-  height: '27px',
-  borderRadius: '4px',
-  border: '1px solid rgba(0,0,0,0.1)',
-  backgroundColor: color,
-  '&:focus': {
-    boxShadow: `0 0 0 1px ${theme.colors.sidebar.background}, 0 0 0 3px ${theme.colors.primary}`,
-  },
-}));
+const Container = styled.button<{ background: string }>(
+  ({ theme, background }) => ({
+    outline: 'none',
+    width: '40px',
+    height: '27px',
+    borderRadius: '4px',
+    border: '1px solid rgba(0,0,0,0.1)',
+    background,
+    '&:focus': {
+      boxShadow: `0 0 0 1px ${theme.colors.sidebar.background}, 0 0 0 3px ${theme.colors.primary}`,
+    },
+  }),
+);
 
 interface Props {
   id?: string;
-  value: Sketch.Color;
+  value: Sketch.Color | Sketch.Gradient;
```

Comment:
Can we do separate components here? `ColorInputField`, `GradientInputField`

Reply:
yes! They look almost the same 🤔 That's why I combined them but I'll separate them